api/v1beta2/modulereleasemeta_types.go

```diff
@@ -0,0 +1,61 @@
+package v1beta2
+
+import (
+	apimetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ModuleReleaseMeta is the representation of the channel-version pairs for modules. Each item represents
+// a module version along with its assigned channel.
+//
+// +genclient
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:singular=modulereleasemeta,path=modulereleasemetas,shortName=mrm
+// +kubebuilder:printcolumn:name="Age",type="date",JSONPath=".metadata.creationTimestamp"
+// +kubebuilder:storageversion
+
+type ModuleReleaseMeta struct {
+	apimetav1.TypeMeta   `json:",inline"`
+	apimetav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec ModuleReleaseMetaSpec `json:"spec,omitempty"`
+}
+
+// ModuleReleaseMetaSpec defines the version-channel assignments for a module
+type ModuleReleaseMetaSpec struct {
+	// ModuleName is the name of the Module.
+	// +kubebuilder:validation:Pattern:=`^([a-z]{3,}(-[a-z]{3,})*)?$`
+	// +kubebuilder:validation:MaxLength:=64
+	ModuleName string `json:"moduleName"`
+
+	// Channels is the list of module channels with their corresponding versions.
+	// +listType=map
+	// +listMapKey=channel
+	Channels []ChannelVersionAssignment `json:"channels"`
+}
+
+// +kubebuilder:object:root=true
+
+// ModuleReleaseMetaList contains a list of ModuleReleaseMeta.
+type ModuleReleaseMetaList struct {
+	apimetav1.TypeMeta `json:",inline"`
+	apimetav1.ListMeta `json:"metadata,omitempty"`
+	Items              []ModuleReleaseMeta `json:"items"`
+}
+
+type ChannelVersionAssignment struct {
+	// Channel is the module channel of the corresponding module version.
```

Comment:
```suggestion
	// Channel is the module channel.
```

I think I would strip the latter part here because this may indicate that a given version can only be used in one channel which is not the case.